src/dagmc/dagmcmetadata.cpp

```diff
@@ -0,0 +1,591 @@
+      exit(EXIT_FAILURE);
+    }
+
+    // set the material value
+    volume_material_property_data_eh[eh] = grp_name;
+
+    // not graveyard or vacuum or implicit compliment
+    if (grp_name.find("Graveyard") == std::string::npos && grp_name.find("Vacuum") == std::string::npos
+        && !(DAG->is_implicit_complement(eh))) {
+      // set the
+      volume_material_data_eh[eh] = material_props[0];
+    }
+    // found graveyard
+    else if (grp_name.find("Graveyard") != std::string::npos) {
+      volume_material_property_data_eh[eh] = "mat:Graveyard";
+      volume_material_data_eh[eh] = "Graveyard";
+    }
+    // vacuum
+    else if (grp_name.find("Vacuum") != std::string::npos) {
+      volume_material_property_data_eh[eh] = "mat:Vacuum";
+      volume_material_data_eh[eh] = "Vacuum";
+    }
+    // implicit complement
+    else if (DAG->is_implicit_complement(eh)) {
+      if (implicit_complement_material == "") {
+        std::cout << "Implicit Complement assumed to be Vacuum" << std::endl;
+        volume_material_property_data_eh[eh] = "mat:Vacuum";
+        volume_material_data_eh[eh] = "Vacuum";
+      } else {
+        volume_material_property_data_eh[eh] = "mat:" + implicit_complement_material;
+        if (implicit_complement_density != "")
+          volume_material_property_data_eh[eh] += "/rho" + implicit_complement_density;
+        volume_material_data_eh[eh] = implicit_complement_material;
+        volume_density_data_eh[eh] = implicit_complement_density;
+      }
+    }
+  }
+}
+
+// parse the importance data from the file
+void dagmcMetaData::parse_importance_data() {
+  std::map<moab::EntityHandle, std::vector<std::string> > importance_assignments;
+  importance_assignments = get_property_assignments("importance", 3, ":");
+
+  int num_vols = DAG->num_entities(3);
+
+  std::vector<std::string> importance_assignment; // importance conditions for the current entity
+  // loop over all volumes set up the generic importance data
+  for (int i = 1; i <= num_vols; ++i) {
+    int volid = DAG->id_by_index(3, i);
+    moab::EntityHandle eh = DAG->entity_by_index(3, i);
+
+    // vector of importance values
+    importance_assignment = importance_assignments[eh];
+
+    // set the value of each string for
+    std::string importances = "|";
+    for (int j = 0 ; j < importance_assignment.size() ; j++) {
+      if (importance_assignment[j] == "")
+        break;
+      // delimit each particle/value pair with a pipe symbol
+      importances += importance_assignment[j] + "|";
+      // also split to get key-value
+      std::pair<std::string, std::string> pair = split_string(importance_assignment[j], "/");
+      // add to the unique collection of particle names
+      imp_particles.insert(pair.first);
+      // insert into map too
+      if (importance_map[eh].count(pair.first) == 0) {
+        importance_map[eh][pair.first] = atof(pair.second.c_str());
```

Comment:
I would recommend using `std::strtod` rather than `atof`, for better handling of `std::string` and error state [1].

[1] https://stackoverflow.com/a/12369101